lib/cocina/models/builders/title_builder.rb

```diff
@@ -125,40 +159,76 @@ def title_from_structured_values(title)
+                         elsif result.present?
+                           "#{result.sub(/[ .,]*$/, '')}. #{part_name_number}. "
+                         else
+                           "#{part_name_number}. "
+                         end
               end
             when 'main title', 'title'
-              structured_title = "#{structured_title}#{value}"
+              result = "#{result}#{value}"
             when 'subtitle'
               # subtitle is preceded by space colon space, unless it is at the beginning of the title string
-              structured_title = if !add_punctuation?
-                                   [structured_title, value].join(' ')
-                                 elsif structured_title.present?
-                                   "#{structured_title.sub(/[. :]+$/, '')} : #{value.sub(/^:/, '').strip}"
-                                 else
-                                   structured_title = value.sub(/^:/, '').strip
-                                 end
+              result = if !add_punctuation?
+                         [result, value].join(' ')
+                       elsif result.present?
+                         "#{result.sub(/[. :]+$/, '')} : #{value.sub(/^:/, '').strip}"
+                       else
+                         result = value.sub(/^:/, '').strip
+                       end
             end
           end
-          structured_title
+          result
         end
-        # rubocop:enable Metrics/MethodLength
         # rubocop:enable Metrics/BlockLength
         # rubocop:enable Metrics/CyclomaticComplexity
+        # rubocop:enable Metrics/MethodLength
+        # rubocop:enable Metrics/PerceivedComplexity
+
+        # main_title is title.structuredValue.value with type 'main title' (or just title.value)
+        # @param [Cocina::Models::Title] title with structured values
+        # @return [String] the main title value
+        #
+        # rubocop:disable Metrics/MethodLength
+        # rubocop:disable Metrics/PerceivedComplexity
+        def main_title_from_structured_values(cocina_title)
+          result = ''
+          # combine pieces of the cocina structuredValue into a single title
+          cocina_title.structuredValue.each do |structured_value|
+            # There can be a structuredValue inside a structuredValue.  For example,
+            #   a uniform title where both the name and the title have internal StructuredValue
+            return main_title_from_structured_values(structured_value) if structured_value.structuredValue.present?
+
+            value = structured_value.value&.strip
+            next unless value
+
+            case structured_value.type&.downcase
+            when 'nonsorting characters'
+              non_sort_value = "#{value}#{non_sorting_padding(cocina_title, value)}"
+              result = if result.present?
+                         "#{non_sort_value}#{result}"
+                       else
+                         non_sort_value
```

Comment:
If `result` is either `nil` or `''`, won't `"#{non_sort_value}#{result}"` be equivalent to `non_sort_value`? The condition check may be unnecessary.

Reply:
good eye!  thanks.